Jellyfin.Server/Migrations/Routines/UpdateDatabaseToTenDotTen.cs

```diff
@@ -0,0 +1,324 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Emby.Server.Implementations.Data;
+using Jellyfin.Database.Implementations.Entities;
+using Jellyfin.Extensions;
+using MediaBrowser.Controller;
+using MediaBrowser.Controller.Library;
+using Microsoft.Data.Sqlite;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Server.Migrations.Routines;
+
+/// <summary>
+/// Updates the library.db to version 10.10.z; replaces the migration code from the old SqliteItemRepository and SqliteUserDataRepository.
+/// </summary>
+[JellyfinMigration("2025-04-19T00:00:00", nameof(UpdateDatabaseToTenDotTen))]
+internal class UpdateDatabaseToTenDotTen : IDatabaseMigrationRoutine
+{
+    private const string DbFilename = "library.db";
+    private readonly ILogger<UpdateDatabaseToTenDotTen> _logger;
+    private readonly IServerApplicationPaths _paths;
+    private readonly IUserManager _userManager;
+
+    public UpdateDatabaseToTenDotTen(
+        ILogger<UpdateDatabaseToTenDotTen> logger,
+        IServerApplicationPaths paths,
+        IUserManager userManager)
+    {
+        _logger = logger;
+        _paths = paths;
+        _userManager = userManager;
+    }
+
+    /// <inheritdoc/>
+    public void Perform()
+    {
+        const string ChaptersTableName = "Chapters2";
+
+        string[] queries =
+        [
+            "create table if not exists TypedBaseItems (guid GUID primary key NOT NULL, type TEXT NOT NULL, data BLOB NULL, ParentId GUID NULL, Path TEXT NULL)",
```

Comment:
at this stage the tables should already exist. We are migrating, so that means its not a new install and on new installs this migration should not run either. If its an actual migration, those tables have to already exist in the first place.